builds/misc/metrics-collector-images-release.yaml

```diff
@@ -4,108 +4,415 @@ pr: none
+
+        - task: PublishBuildArtifacts@1
+          displayName: 'Publish Artifacts'
+          inputs:
+            PathtoPublish: '$(Build.BinariesDirectory)/publish'
+            ArtifactName: 'core-windows'
+
+        - template: templates/image-windows.yaml
+          parameters:
+            name: Metrics Collector (customer-facing)
+            imageName: azureiotedge-metrics-collector
+            project: Microsoft.Azure.Devices.Edge.Azure.Monitor
+            version: $(version)
+
+        # for E2E tests only
+        - template: templates/image-windows.yaml
+          parameters:
+            imageName: azureiotedge-agent
+            name: "Edge Agent"
+            project: Microsoft.Azure.Devices.Edge.Agent.Service
+            version: $(version)
+            buildx_flag: 'true'
+            use_rocksdb: true
+        - template: templates/image-windows.yaml
+          parameters:
+            imageName: azureiotedge-hub
+            name: "Edge Hub"
+            project: Microsoft.Azure.Devices.Edge.Hub.Service
+            version: $(version)
+            buildx_flag: 'true'
+            use_rocksdb: true
+        - template: templates/image-windows.yaml
+          parameters:
+            imageName: azureiotedge-direct-method-sender
+            name: "Direct Method Sender"
+            project: DirectMethodSender
+            version: $(version)
+            buildx_flag: 'true'
+
 ################################################################################
-  - job: MetricsCollector_Linux
+- stage: PublishManifests
 ################################################################################
-    displayName: Metrics Collector Linux Build
-    timeoutInMinutes: 30
-    pool:
-      name: $(pool.linux.name)
-      demands:
-        - ImageOverride -equals agent-aziotedge-ubuntu-20.04-msmoby
-    steps:
+  displayName: Publish Manifest Images
+  pool:
+    name: $(pool.linux.name)
+    demands:
+      - ImageOverride -equals agent-aziotedge-ubuntu-20.04-msmoby
+  jobs:
+    - job:
+      steps:
       # If you need to test this job. Create a new Service Connection in the Project Settings (Azure DevOps)
       # and update the pipeline variable.
       - task: Docker@2
         displayName: Docker login edgerelease
         inputs:
           command: login
           containerRegistry: $(service.connection)
+      - script: scripts/linux/buildManifest.sh -r $(registry.address) -v $(version) -t $(System.DefaultWorkingDirectory)/edge-modules/azure-monitor/docker/manifest.yaml.template -n microsoft --tags "$(tags)"
+        displayName: 'Publish Metrics Collector Manifest'
 
-      - script: scripts/linux/buildBranch.sh -c $(Build.Configuration) --no-rocksdb-bin
-        name: build
-        displayName: Build ($(Build.Configuration))
+################################################################################
+- stage: RunE2ETest
+################################################################################
```

Comment:
Given how `RunE2ETest` are being used across pipelines, please turn it into a YAML template and reuse them across pipelines.

Reply:
As stated in the review description, things have been done this way with intent.

- It's not a straight copy-paste. RunE2ETest may be used across pipelines, but the templates called are different in different cases. For instance, e2e-run-metrics-collector.yaml is called only in this YAML and will never be called in any other YAML.
- This means parameter passing of some sort; this is the kind of thing we should think about and discuss before a large(r) refactor -- and there is a work item tracked to do just this.
- Also, this Metrics Collector YAML is self-contained. We don't have the same concerns here that we do in our release process where we're not sure if the same code is running in both Build Images and Images Release. In addition, we already have conversations ongoing about deprecating this version of the Metrics Collector.
- Expending sub-optimal engineering effort to figure out a pattern that fits all pipelines here, on something that we may deprecate soon (and is a container anyway) is not a great idea.

While your suggestion is very valid in a larger context, I shan't be taking it here.